### ISSUE TITLE
Prebid Test Bidder

### ISSUE THREAD

**Issue by gglas** [prebid/Prebid.js#7365]
It would be great if Prebid had an example bidder that replied with a static creative to function as a participant in the auction that always replies with something.

The details of this configuration get complicated relatively quickly, considerations include (but wouldn't be limited to) : 

Bid Price
Format
Consent status 

This would be statically hosted, and provide a reliable way to ensure there are bids in a given prebid auction for testing purposes.

**Comment by patmmccann** [comment]
@dgirardi this feature request is slightly diff that what is solved in 7801; the idea is to have an endpoint that returns a creative that can be used in place of the appnexus endpoint on the examples page

**Comment by dgirardi** [comment]
@patmmccann true, but the assumption is that if a publisher has to include and configure a "test bidder" module to make the examples work, they could also include and configure a debugging module that has the same effect.

Maybe a good middle ground is to have the debugging module automatically intercept all requests for say `testBidder` ?

**Comment by patmmccann** [comment]
I dont follow how the debugging module solves the three or four issues we consolidated into this one. I think publishers want to demonstrate end to end a bid was sent out and markup was returned and rendered. The interception seems to almost by definition break the end to end test.

**Comment by dgirardi** [comment]
@patmmccann I don't disagree, but if the goal is true end-to-end, wouldn't it be easier to strike a deal with some member SSPs to get some demo account set up that always replies with some bid? anything else is a tradeoff between maintenance costs and true "real-world-ness".

**Comment by muuki88** [comment]
We are currently starting to add more prebid native bidders and would like to check if the native designs work as expected with GAM as well and also to demonstrate them SSPs.

I couldn't figure exactly if the _debugging module_ is suitable to do this? The replacing functions look very generic, which means I would only reverse engineer the proper fields required in the `bidResponse` object so that native works, right?

**Comment by Hamper** [comment]
I think that testBidder can be very usable for something like native design testing or video tests, it can looks like
```js
{
  bidder: 'test',
  parameters: {
    type: 'native',
    bid: { /* full bid response */ }
  }
}
```
and can contains some minimal tests like check that type match available types for adunit

**Comment by patmmccann** [comment]
If we build a new bid adapter called 'notForProdExampleBidder' that was not on the download page but was always in not for prod giant build for examples pages, and that bidder did not generate network traffic but instead had a json forcing a bid where the markup was an <img> call to something on imgur or somewhere like that, we could solve this. Or perhaps mockBidAdapter.js

**Comment by patmmccann** [comment]
As an alternative, we're preferring some sort of config that could be identified in an example that would abstract away much of the work of mocking responses

**Comment by patmmccann** [comment]
potential creative for mockBidAdapter.js to always respond with in adm 
```

<a href="https://prebid.org" target="_top"><img width="300" height="250" style="border-style: none" src="https://vcdn.adnxs.com/p/creative-image/27/c0/52/67/27c05267-5a6d-4874-834e-18e218493c32.pg"></a>
```

**Comment by patmmccann** [comment]
let's not bother with network traffic for the image, lets hard code it in base64 in thiis manner: 

```
<img style='display:block; width:100px;height:100px;' id='base64image'
       src='data:image/jpeg;base64, LzlqLzRBQ... <!-- Base64 data -->' />

```
